Makefile

```diff
@@ -1585,16 +1587,12 @@ lint-ci: lint-js-ci lint-cpp lint-py lint-md lint-addon-docs lint-yaml-build lin
 		echo "Conflict marker detected in one or more files. Please fix them first." >&2 ; \
 		exit 1 ; \
 	fi
-else
-lint lint-ci:
-	$(info Linting is not available through the source tarball.)
-	$(info Use the git repo instead: $$ git clone https://github.com/nodejs/node.git)
-endif
```

Comment:
I wonder if this is going to be disruptive for tarball users, as the lint step could now fail if e.g. they don't  have internet access.

Reply:
I don't think it'll be disruptive. 

For starters, you need internet to download the tarball, so it's okay to assume that internet is needed to perform advanced operations.

Secondly, the linting of other parts, such as the python and YAML linter also require internet, so it's not un-common for internet to be needed to lint.

Reply:
> For starters, you need internet to download the tarball, so it's okay to assume that internet is needed to perform advanced operations.

I don't see the logic here, one can certainly get the tarball without the internet, and even if they machine had access to the internet in the past, it seems careless to assume it would still have it after the tarball has been extracted and .the user starts working with it.

> Secondly, the linting of other parts, such as the python and YAML linter also require internet, so it's not un-common for internet to be needed to lint.

Sure, but before your PR, `make lint` would still exit with code 0 without the internet, it would no longer be the case after your PR. Also, this can be disruptive not just to folks who lack internet access, it changes the behavior of `make lint` quite a lot.

After verification, `lint` and `lint-ci` are not targets of any other job, so it would seem like it'd be disruptive only for folks who were running `make lint` despite the fact it would do nothing but print a warning. Let's maybe not backport it too far to avoid disrupting automated workflows out there.

Reply:
> I don't think it'll be disruptive.
> 
> For starters, you need internet to download the tarball, so it's okay to assume that internet is needed to perform advanced operations.
> 

No, you're are supposed to be able to build Node.js from the source tarball without having to download anything else during the build, which is why this definition of `lint-ci` exists and should not be removed.

Reply:
Oh, good to know. I don't know too much about tarball + network interactions.

I would like to point out, `lint-ci` requires internet to complete, as it runs `lint-yaml-build`. However, before this PR, `lint` did not require internet access.

Reply:
To address the concern raised here, we need a way to make sure `make lint` and `make lint-ci` keep the same behavior in tarballs, i.e. require no internet access and complete with code 0 printing the same warnings.

Reply:
`lint-ci` already requires internet to complete, but if you have any ideas to resolve this concern for `lint`, I'm all ears. I could have it not fail if `lint-js-build` fails, but instead exit 0 early with a warning that the build of `lint-js` failed?

Reply:
`lint-ci` does not require internet access:

```console
$ curl https://nodejs.org/dist/latest/node-v22.6.0.tar.xz | tar -xJ
$ # disconnect from the internet
$ cd node-v22.6.0 
$ make lint-ci
Linting is not available through the source tarball.
Use the git repo instead: $ git clone https://github.com/nodejs/node.git
make: Nothing to be done for 'lint-ci'.
$ echo $?
0
```

Reply:
Ahh, I got confused, my apologies. You are correct.

Reply:
I tried the following with an empty npm cache and without internet connection:

```console
$ rm -rf tools/eslint/node_modules    
$ touch tools/eslint/package-lock.json
$ make lint                      
cd tools/eslint && if [ -x ""/Users/duhamean/Documents/node/node"" ] && [ -e ""…/node"" ]; then ""…/node"" …/./deps/npm/bin/npm-cli.js ci; elif [ -x `command -v node` ] && [ -e `command -v node` ] && [ `command -v node` ]; then `command -v node` …/./deps/npm/bin/npm-cli.js ci; else echo "No available node, cannot run \"node …/./deps/npm/bin/npm-cli.js ci\""; exit 1; fi;
npm warn cli npm v10.8.2 does not support Node.js v23.0.0-pre. This version of npm supports the following node versions: `^18.17.0 || >=20.5.0`. You can find the latest version at https://nodejs.org/.
⠧
```

The command never completes (arguably a bug with npm), but anyway it would have downloaded stuff from the internet if there was a working connection, which is not what we want in tarballs.
I don't understand why you do not implement what https://github.com/nodejs/node/pull/54231/commits/d3fa8b0c35897f2298ef2434e17c262d730fe250#r1711379416 asked, i.e. keep the current target for `lint` and `lint-ci` in tarballs.